docs/DesignDocs/AttestationApiRequirements.md

```diff
@@ -0,0 +1,429 @@
+an implementation acts in one or more of the three roles: Attester,
+Verifier, or Relying Party.  An Attester generates signed Evidence
+to be processed by a Verifier.  A Verifier appraises Evidence using
+an Appraisal Policy (and often Endorsements as well), and generates
+Attestation Results to be processed by a Relying Party.  A Relying
+Party appraises Attestation Results and typically uses them for
+some authorization decision.
+
+The APIs are thus centered around the conceptual data flow
+depicted in Figure 1 in the
+[RATS Architecture](https://tools.ietf.org/html/draft-ietf-rats-architecture-02#page-8).
+
+Specific formats of Evidence, Attestation Results, etc. are the
+responsibility of the relevant plugins to implement, not the application.
+Similarly, plugins are responsible for TEE-specific details.
+
+The (non-attestation specific) protocol between the Attester and
+a Relying Party is the responsibility of the application, not
+the OE SDK or plugins.
+
+The (attestation-specific) protocol between a Verifier and either
+an Attester (in the Passport model) or a Relying Party (in the
+Background-check model) is the responsibility of the relevant
+plugins to implement, not the application.
+
+It is an open question whether it is the responsibility of the app,
+the OE SDK, or plugins to actually apply an appraisal policy (e.g., to do
+the actual verify operation).  The appraisal policy might be hard-coded,
+or be encapsulated inside endorsements, or be configured separately
+through other APIs, or any combination of those.  This draft will cover
+APIs that might be needed in any of these designs: appraisal in the app,
+appraisal in the SDK, and appraisal in a plugin.  The open question
+is which one(s) to support, and hence which subset of abstract APIs
+are needed.
+
+Note that the deprecated `oe_verify_report()` API was designed for a
+combined Verifier+Relying Party, and did the appraisal underneath the
+API (i.e., in the OE SDK).  However, convoluting the two roles does
+not meet the more general requirements stated, and in separating them,
+the open issue remains.
+
+## Specification
+
+The APIs below are described as _abstract_ APIs, and any concrete
+APIs in C (or any other language) needs to accommodate an equivalent
+set of operations and support for an (at least) equivalent set of inputs
+and outputs.  This means the actual mapping of functions or arguments
+need not be 1:1 as long as a mapping exists.
+
+These APIs apply to both the "northbound" API exposed to apps by the SDK,
+as well as the "southbound" API between the SDK and plugins.
+
+### Attester APIs
+
+#### GetEvidence call
+
+This API is used by an app to retrieve an evidence buffer that can
+be sent to a Verifier (in the Passport model), or a Relying Party
+(in the Background-check model).
+
+Inputs:
+
+* Requested Evidence format, if any (e.g., CWT, JWT, X.509, etc.).
+  Defaults to letting the plugin choose one.
+* Challenge buffer OCTET STRING
```

Comment:
Does it make sense to add a MAX limit for all the different buffers (challenge buffer, evidence buffer, etc)?

Reply:
I don't know.